src/imitation/util/video_wrapper.py

```diff
@@ -61,18 +72,27 @@ def _reset_video_recorder(self) -> None:
                 metadata={"episode_id": self.episode_id},
             )
 
-    def reset(self):
+    def reset(self, **kwargs):
+        new_obs = super().reset(**kwargs)
         self._reset_video_recorder()
         self.episode_id += 1
-        return self.env.reset()
+        return new_obs
 
     def step(self, action):
-        res = self.env.step(action)
+        obs, rew, done, info = self.env.step(action)
         self.video_recorder.capture_frame()
-        return res
+        # is it crazy to save the video path at every step?
+        info["video_path"] = self.get_current_video_path()
```

Comment:
Probably best to only store this on the first step

Reply:
The fragmenter will splice trajectories into multiple fragments, so storing it once per episode may not be enough (unless you make the fragmenter cleverer, and have it propagate the video path as needed). You can avoid extra memory by making sure `self.get_current_video_path()` returns the same object on successive calls if underlying path hasn't changed (some kind of memoization should work).

Reply:
I think the issue I was worried about was that the trajectories would be stored on disk, in which case the video path would be duplicated a lot. Although maybe pickle is smart enough to make it a reference if it's the same object? IIRC `str` is immutable so it probably gets stored on disk as a value but `Path` probably won't be